api.d.ts

```diff
@@ -1113,8 +1113,56 @@ export interface ILexerConfig {
      * Note that this would have negative performance implications.
      */
     safeMode?: boolean
+
+    /**
+     * A custom error message provider.
+     * Can be used to override the default error messages.
+     * For example:
+     *   - Translating the error messages to a different languages.
+     *   - Changing the formatting.
+     */
+    errorMessageProvider?: ILexerErrorMessageProvider
 }
 
+export interface ILexerErrorMessageProvider {
+    /**
+     * Unexpected Character Error happens when lexer could not find token match for input text
```

Comment:
Maybe something like?
> An Unexpected Character Error occurs when the lexer is unable to match a range of  one or more 
   characters in the input text against any of the Token Types in it's Lexer definition.